.github/workflows/iroha1.yml

```diff
@@ -0,0 +1,113 @@
+name: Iroha 1 branch workflow
+on:
+  push:
+    branches:
+      - iroha1-main
+    tags:
+      - '*'
```

Comment:
```suggestion
  push:
    tags:
      - 'v1*'
```

Reply:
we should differentiate between publishing a release and merging into `main`

Reply:
releases should **only** be built from tags

Reply:
The scope of this PR is to fix the branch name and the workflow file name (to avoid confusion), the rest I left as it was

Reply:
ok, but we should split our CI into these workflows:
1. workflow that runs on every merge to `main`
2. workflow that runs on every merge to `iroha1-main`
3. workflow that runs on every `v1*` tag
4. workflow that runs on every `v2*` tag

I think these should all be separate CI files, but maybe you can make them into 2 files